### DISCUSSION TITLE
Micropython ESP32-S3 ProS3 library install

### DISCUSSION THREAD

**Discussion by b-palaniappan** [micropython/micropython#9680] [Libraries & Drivers] [answered]
I got a new ESP32-S3 ( UM ProS3) and installed latest micropython v1.19.1.
I am trying to install `aioble` and had problem with not having `mip` module in it. So I installed it thru `mpremote`,

```bash
❯ mpremote mip install aioble
Install aioble
Installing aioble (latest) from https://micropython.org/pi/v2 to /lib
Installing: /lib/aioble/__init__.mpy
Installing: /lib/aioble/core.mpy
Installing: /lib/aioble/device.mpy
Installing: /lib/aioble/peripheral.mpy
Installing: /lib/aioble/server.mpy
Done
```
Seems the install was successful. But after the import, when I run, `import aioble` I am getting error,
```bash
>>> import aioble
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: incompatible .mpy file
```
Not sure what I am doing wrong. Any help is appreciated.

**Comment by jimmo** [comment, marked as answer]
> not having mip module in it

mip will be in the next release.

> install latest micropython v1.9.1.

Thanks for reporting this! I have just fixed the underlying bug -- https://github.com/micropython/micropython/pull/9681. We will need to merge this and update the mip index.

In the meantime, you can either:
 - Update the board to the latest version (i.e. a recent nightly/unstable build -- despite the name, don't expect any problems)
 - Install the python sources instead of the .mpy files. `mpremote mip install --no-mpy aioble`. You should `mpremote rm /lib/aioble/__init__.mpy` (and the others) first.

**Reply by b-palaniappan** [reply]
I went with option one. Erased flash and flashed latest nightly,
```
MicroPython v1.19.1 on 2022-10-14; ProS3 with ESP32-S3
```
and installed `aioble` as `.mpy` and it works. 
Thanks @jimmo for your quick fix and response. I appreciate your response and quick fix.

**Reply by dpgeorge** [reply]
This is now fixed in 1ba0e8ff96334af986d2b7d90f6d86af27595d28
